tools/nvcc_fix_deps.py

```diff
@@ -0,0 +1,118 @@
+"""Tool to fix the nvcc's dependecy file output
+
+Usage: python nvcc_fix_deps.py nvcc [nvcc args]...
+
+This wraps nvcc to ensure that the dependency file created by nvcc with the
+-MD flag always uses absolute paths. nvcc sometimes outputs relative paths,
+which ninja interprets as an unresolved dependency, so it triggers a rebuild
+of that file every time.
+
+The easiest way to use this is to define:
+
+CMAKE_CUDA_COMPILER_LAUNCHER="python;tools/nvcc_fix_deps.py;ccache"
+
+"""
+
+import sys
+import subprocess
+from pathlib import Path
+from typing import List, TextIO, Optional
+
+def resolve_include(path: Path, include_dirs: List[Path]) -> Path:
+    for include_path in include_dirs:
+        abs_path = include_path / path
+        if abs_path.exists():
+            return abs_path
+
+    paths = '\n    '.join(str(d / path) for d in include_dirs)
```

Comment:
Nit (although it goes against my gut feeling, but I was told by many PyTorch contributors that the like to following this rule: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings )
```suggestion
    paths = "\n    ".join(str(d / path) for d in include_dirs)
```